src/main/java/core/basesyntax/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.handler;
+
+import core.basesyntax.db.Storage;
+
+public class PurchaseOperationHandler implements ShopOperationHandler {
+    @Override
+    public void doOperation(String fruitName, String quantityToOperate) {
+        Integer oldQuantityValue = Storage.FRUIT_STORAGE.get(fruitName.toLowerCase());
+        Storage.FRUIT_STORAGE.put(fruitName.toLowerCase(),
+                oldQuantityValue - Integer.parseInt(quantityToOperate));
```

Comment:
After above changes solution should work properly if file has standard order of operations (where balance is first)
But I'd not recommend to rely on proper input and add additional checks.
First is here not to fall with NPE for selling fruit that doesn't exist or end up with negative amount:
```suggestion
        Integer oldQuantityValue = Storage.FRUIT_STORAGE.get(fruitName.toLowerCase());
        if (oldQuantityValue == null || oldQuantityValue < quantityToOperate) {
            throw new IllegalArgumentException("Available quantity of " + fruitName 
                    + " is less than required for purchase operation");
        }
        Storage.FRUIT_STORAGE.put(fruitName.toLowerCase(),
                oldQuantityValue - Integer.parseInt(quantityToOperate));
```

Reply:
Done